[🐛 Bug]: Node 20: Errors with custom commands and global namespace

### Have you read the Contributing Guidelines on issues?

- [X] I have read the [Contributing Guidelines on issues](https://github.com/webdriverio/webdriverio/blob/main/CONTRIBUTING.md#reporting-new-issues).

### WebdriverIO Version

8.16.11

### Node.js Version

20.7.0

### Mode

WDIO Testrunner

### Which capabilities are you using?

_No response_

### What happened?

After updating the node version from version 18 to version 20 and installing the next/canary release of the @wdio/cli package (8.16.12-alpha.3), I am having some error with the namespaces

![image](https://github.com/webdriverio/webdriverio/assets/132624270/9e38d2eb-d815-4550-a745-a39c4fe42384)

This is how I have the vars declared in the namespace:

![image](https://github.com/webdriverio/webdriverio/assets/132624270/23bdad49-2b05-40ae-a131-55052d1f2d4b)

![image](https://github.com/webdriverio/webdriverio/assets/132624270/65249259-7b3b-474d-8c6f-3afbfcebbd9f)


### What is your expected behavior?

_No response_

### How to reproduce the bug.

[sample_project.zip](https://github.com/webdriverio/webdriverio/files/12686094/sample_project.zip)


### Relevant log output

```typescript
PS C:\Repos\QA\e2e-dealerk-dashboard-manager> npm run test:it

> e2e-dashboard-user@1.0.0 test:it
> (set country=it || export country=it) && wdio ./config/wdio.local.ts "--suite" it "--cucumberOpts.tags" "@test"

2023-09-21T14:19:41.143Z ERROR @wdio/config:ConfigParser: Failed loading configuration file: file:///C:/Repos/QA/e2e-dealerk-dashboard-manager/config/wdio.local.ts: ⨯ Unable to compile TypeScript:
src/support/hooks.ts:36:16 - error TS7017: Element implicitly has an 'any' type because type 'typeof globalThis' has no index signature.

36         global.startScenario = new Date(Date.now() - 1000);
                  ~~~~~~~~~~~~~
src/support/hooks.ts:44:45 - error TS2339: Property 'isVisualTesting' does not exist on type 'Browser'.

44         if (result.passed || (await browser.isVisualTesting()))
                                               ~~~~~~~~~~~~~~~
src/support/hooks.ts:49:16 - error TS7017: Element implicitly has an 'any' type because type 'typeof globalThis' has no index signature.

49         global.endScenario = new Date(Date.now() + 2000);
                  ~~~~~~~~~~~
src/support/hooks.ts:51:64 - error TS7017: Element implicitly has an 'any' type because type 'typeof globalThis' has no index signature.

51             .replace("@TIMESTAMP_START", JSON.stringify(global.startScenario.getTime()))
                                                                  ~~~~~~~~~~~~~
src/support/hooks.ts:52:62 - error TS7017: Element implicitly has an 'any' type because type 'typeof globalThis' has no index signature.

52             .replace("@TIMESTAMP_END", JSON.stringify(global.endScenario.getTime()))}`, "text/plain");
                                                                ~~~~~~~~~~~

src/support/hooks.ts:36:16 - error TS7017: Element implicitly has an 'any' type because type 'typeof globalThis' has no index signature.

                  ~~~~~~~~~~~~~
src/support/hooks.ts:44:45 - error TS2339: Property 'isVisualTesting' does not exist on type 'Browser'.

44         if (result.passed || (await browser.isVisualTesting()))
                                               ~~~~~~~~~~~~~~~
src/support/hooks.ts:49:16 - error TS7017: Element implicitly has an 'any' type because type 'typeof globalThis' has no index signature.

49         global.endScenario = new Date(Date.now() + 2000);
                  ~~~~~~~~~~~
src/support/hooks.ts:51:64 - error TS7017: Element implicitly has an 'any' type because type 'typeof globalThis' has no index signature.

51             .replace("@TIMESTAMP_START", JSON.stringify(global.startScenario.getTime()))
                                                                  ~~~~~~~~~~~~~
src/support/hooks.ts:52:62 - error TS7017: Element implicitly has an 'any' type because type 'typeof globalThis' has no index signature.

52             .replace("@TIMESTAMP_END", JSON.stringify(global.endScenario.getTime()))}`, "text/plain");
                                                                ~~~~~~~~~~~

PS C:\Repos\QA\e2e-dealerk-dashboard-manager>
```


### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Is there an existing issue for this?

- [X] I have searched the existing issues

--- a/website/docs/TypeScript.md
+++ b/website/docs/TypeScript.md
@@ -123,6 +123,27 @@ const config: Options.WebdriverIO = {
 }
 ```
 
+## Missing Types
+
+When using Node 20 or above, wdio runs ts-node with different settings as this is required in order to keep things running.
+These settings can cause your custom types not to be loaded, if this happens there are a few ways you can fix this, of which the easiest I will show below.
+
+Using ts-node's environment variables
+```
+TS_NODE_FILES=true wdio run ./wdio.conf.ts
+```
+
+Using tsconfig
+```
+{
+  "ts-node": {
+    "files": true
+  },
+}
+```
+
+For more information checkout the (ts-node documentation)[https://typestrong.org/ts-node/docs/troubleshooting#missing-types].
+
 ## Tips and Hints
 
 ### Compile & Lint

